From 5181935bc1cec67af03b97866b61ad7bafce4e29 Mon Sep 17 00:00:00 2001
From: Yumeko Margatroid <ChildOfDreams@users.noreply.github.com>
Date: Tue, 23 Jul 2024 23:31:34 +0100
Subject: [PATCH] racing: fix sendPlaces when everyone has finished

---
 game/cars/racing/DistributedRaceAI.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/game/cars/racing/DistributedRaceAI.py b/game/cars/racing/DistributedRaceAI.py
index 14aeb2e..d5329a8 100644
--- a/game/cars/racing/DistributedRaceAI.py
+++ b/game/cars/racing/DistributedRaceAI.py
@@ -97,6 +97,11 @@ class DistributedRaceAI(DistributedDungeonAI):
             playerSegment = self.playerIdToSegment.get(player)
             playerLapsAndSegmentsIds[player] = (playerLap, playerSegment.id)
 
+        if len(playerLapsAndSegmentsIds) == 0:
+            # All players have finished. Just send the update now.
+            self.sendUpdate('setPlaces', [self.places])
+            return
+
         playersOnFurthestLap: List[int] = []
         playersInFurthestSegment: List[int] = []
 
-- 
2.45.2.windows.1

